tests/infra_test.go

```diff
@@ -783,28 +784,39 @@ var _ = Describe("[Serial]Infrastructure", func() {
 			}
 		})
 
-		It("[test_id:4146]should include VMI phase metrics for few running VMs", func() {
-			// this tests requires at least two running VMis. To ensure this condition,
-			// the simplest way is just always run an additional VMI.
-			By("Creating another VirtualMachineInstance")
+		Context("Creating another VirtualMachineInstance", func() {
+			var vmi *v1.VirtualMachineInstance
+			BeforeEach(func() {
+				// this tests requires at least two running VMis. To ensure this condition,
+				// the simplest way is just always run an additional VMI.
+				// `pod` is the pod of the virt-handler of the node on which we run all the VMIs
+				// when setting up the tests. So we implicitely run all the VMIs on the same node,
+				// so the test works. TODO: make this explicit.
+				preferredNodeName := pod.Spec.NodeName
+				vmi = pinVMIOnNode(tests.NewRandomVMI(), preferredNodeName)
+				nodeName := startVMI(vmi)
+				Expect(nodeName).To(Equal(preferredNodeName), "Should run VMIs on the same node")
+			})
 
-			// `pod` is the pod of the virt-handler of the node on which we run all the VMIs
-			// when setting up the tests. So we implicitely run all the VMIs on the same node,
-			// so the test works. TODO: make this explicit.
-			preferredNodeName := pod.Spec.NodeName
-			vmi := pinVMIOnNode(tests.NewRandomVMI(), preferredNodeName)
-			nodeName := startVMI(vmi)
-			Expect(nodeName).To(Equal(preferredNodeName), "Should run VMIs on the same node")
+			AfterEach(func() {
+				virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})
```

Comment:
You are not checking if this fails.

Reply:
it was because i didnt check its nil , so it can fail, will add both but then it wont be unconditional

Reply:
Maybe I miss something, but how is unconditional relevant here? You do not have multiple resources to clean.

Reply:
i dont check if vmi != nil
so it can fail , and even panic if i deref a nil ptr

Reply:
So if `vmi` is `nil` and you do do `vmi.Name` you expect this do just panic? I think you interpret unconditional wrongly.

Reply:
i saw we dont really care for secondary asserts, for example we dont nil the ptr.
anyway i will add it